### ISSUE TITLE
"Collapse" of regularization image class in fine-tuned models

### ISSUE THREAD

**Issue by robertsehlke** [XavierXiao/Dreambooth-Stable-Diffusion#5]
I did two fine-tunings of the model:

1. Starting from `sd-v1-4-full-ema.ckpt`, as in the README. 
Training data were photos/selfies. I modified `utils.py` to use a custom identifier. The class noun was "man".

2. Starting from result of (1).
Training data were photos of my partner, again with a custom identifier, but this time using "woman" as the class noun.

The good news are that both models produce pretty high quality results for the new concept. These are not always ideal for photos, I noticed e.g. glossy edges and other artifacts in a number of them, but not too out of the ordinary for SD. However, great generalization to painted styles and pretty good editability.

However, when trying to generate the second set of regularization images from the first finetuned model, I did notice some degradation of the results (I ended up generating regularization images with the original model). The pattern seems similar to the examples without prior-preservation loss from the paper (Fig. 13):

### First round of regularization images - "a photo of a man"

![download](https://user-images.githubusercontent.com/12269260/189326058-5e044354-0c22-41f3-8141-ee018fbbed38.jpg)

### Second round (same prompt) - using ckpt from (1)

![download](https://user-images.githubusercontent.com/12269260/189326142-5aed4d97-bb6a-420e-b82b-89e893ca52d3.jpg)

### Third round (same prompt) - using ckpt from (2)

![download](https://user-images.githubusercontent.com/12269260/189326289-b112c3fa-9fe7-4ff1-9b38-47c65d799e49.jpg)

Not sure why output quality and diversity for "photo of a man" is further degraded here, given the different class noun for the second training.

**Comment by robertsehlke** [comment]
Not too surprisingly "man" and "woman" classes seem to be entangled. Here is what I get for "a photo of a woman" with the three checkpoints:

(original)
![download](https://user-images.githubusercontent.com/12269260/189330702-9fdea7a1-aab4-41fd-9510-62640ea8fcbb.jpg)

(1)
![download](https://user-images.githubusercontent.com/12269260/189330762-28856d09-4ab0-4ed6-bcb0-49d9be3c0f9d.jpg)

(2)
![download](https://user-images.githubusercontent.com/12269260/189330799-30915a4e-619b-434c-aecc-d2d45a6e880b.jpg)

**Comment by XavierXiao** [comment]
Well I think indeed you should use the original model for producing reg images, as fine tuning will do some mysterious thing on the model (remember the model is also fine-tuned on the reg images). I have some thoughts: given that SD cannot generate realistic photo of human, why not use some random images of human you obtained online for regularization? Maybe you can try that as well. I feel like the original SD model will always produce some black-white faint human images with prompt like "photo of a man/woman", so maybe using external, diverse set of human photos serves as better regularization.

**Comment by robertsehlke** [comment]
The first round of regularization images (from the untuned model) are pretty good/usable, so that should be fine.

I was just wondering why generating images with the regularization class noun _after_ fine-tuning leads to such strong drift/collapse for the noun, also since you mentioned in the readme that it looks like they generate regularization images on the fly in the paper. 

I'll try using more regularization images. Upon reading more closely the paper does mention that 
`∼200 × N “a [class noun]” samples are generated, with N being the size of the subject dataset`. So we're looking at 800-1000 recommended.

**Comment by robertsehlke** [comment]
I've now tried it with more regularization images (~300, mix of curated images from the original model + internet photos, at the default 800 steps) - it seems to help a little bit with preserving diversity, but the class prior is still degraded.

Pretty impressive how well the ad hoc regularization works to generate/edit one intended new concept, but this issue limits it a bit. Not sure if they completely solved it in the Dreambooth paper either (though they're clearly aware) or just staved it off with far more regularization images.
